internal/app/machined/internal/phase/kubernetes/remove_all_pods.go

```diff
@@ -47,6 +47,25 @@ func (task *RemoveAllPods) standard() (err error) {
 	// nolint: errcheck
 	defer client.Close()
 
+	// We remove pods with POD network mode first so that the CNI can perform
+	// any cleanup tasks. If we don't do this, we run the risk of killing the
+	// CNI, preventing the CRI from cleaning up the pod's netwokring.
+
+	if err = RemovePodsWithNetworkMode(client, runtimeapi.NamespaceMode_POD); err != nil {
+		return err
+	}
+
+	// With the POD network mode pods out of the way, we kill the remaining
+	// pods.
+
+	if err = RemovePodsWithNetworkMode(client, runtimeapi.NamespaceMode_NODE); err != nil {
```

Comment:
should we kill really all the remaining pods here? (ignoring network namespace altogether?)

Reply:
I mean in the future there might be more modes available, or something else...

Reply:
We should, just didn't know a good way to indicate that all modes should be killed. Maybe `-1` or something?

Reply:
One easy way imho is to use pointer receiver and skip the check if value is `nil`, but probably passing pointer to value would be challenging

Reply:
Went with making `mode` variadic. If there are no modes then all modes are valid.